include/sparrow/array_data.hpp

```diff
@@ -282,17 +284,25 @@ namespace sparrow
     }
 
     template <class D1, class D2>
-    std::strong_ordering operator<=>(const reference_proxy_base<D1>& lhs, const reference_proxy_base<D2>& rhs)
+    OrderingType_t<D1, D2> operator<=>(const reference_proxy_base<D1>& lhs, const reference_proxy_base<D2>& rhs)
     {
         const D1& dlhs = lhs.derived_cast();
         const D2& drhs = rhs.derived_cast();
-        return (dlhs && drhs) ? (dlhs.value() <=> drhs.value()) : (dlhs.has_value() <=> drhs.has_value());
+        if (dlhs && drhs)
+        {
+            return dlhs.value() <=> drhs.value();
+        }
+        return dlhs.has_value() <=> drhs.has_value();
     }
 
     template <class D, not_ref_proxy T>
-    std::strong_ordering operator<=>(const reference_proxy_base<D>& lhs, const T& rhs)
+    std::partial_ordering operator<=>(const reference_proxy_base<D>& lhs, const T& rhs)
```

Comment:
Has this been relaxed for floating types?

Reply:
Yes it is